components/subscription-cleanup-job/internal/cloudprovider/gcp.go

```diff
@@ -0,0 +1,12 @@
+package cloudprovider
+
+type gcpResourceCleaner struct {
+}
+
+func NewGCPeResourcesCleaner(secretData map[string][]byte) ResourceCleaner {
+	return &azureResourceCleaner{}
```

Comment:
I'm not sure about that. I think you meant just to return `&gcpResourceCleaner {}` here instead of `azureResourceCleaner`. It is not supported anyway and will return `nil` at the end but still it is confusing.